### PR TITLE
Add configurable dashboard templates

### PR DESCRIPTION
**Pull request by stphnpt**
### Changelog

```markdown
### Added
- Add configurable dashboard templates (compose, compose_container_view)

### CODE REVIEW

File: Controller/DashboardAdminController.php [hunk 1]
```diff
@@ -38,6 +38,8 @@ public function composeAction(Request $request = null)
             throw new AccessDeniedException();
         }
 
+        $template = $this->admin->getTemplate('compose');
```
Comment (greg0ire) [inline Controller/DashboardAdminController.php]: Don't use One-Time variables and put them faaaar from where they are used unless you have a good reason.

File: Controller/DashboardAdminController.php [hunk 2]
```diff
@@ -56,7 +58,7 @@ public function composeAction(Request $request = null)
 
         $csrfProvider = $this->get('form.csrf_provider');
 
-        return $this->render('SonataDashboardBundle:DashboardAdmin:compose.html.twig', array(
+        return $this->render($template, array(
```
Comment (greg0ire) [inline Controller/DashboardAdminController.php]: Write `return $this->render($this->admin->getTemplate('compose'), array` instead

File: Controller/DashboardAdminController.php [hunk 3]
```diff
@@ -77,6 +79,8 @@ public function composeContainerShowAction(Request $request = null)
             throw new AccessDeniedException();
         }
 
+        $template = $this->admin->getTemplate('compose_container_show');
```
Comment (greg0ire) [inline Controller/DashboardAdminController.php]: Same here

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -46,6 +46,13 @@ public function getConfigTreeBuilder()
                 ->defaultValue('sonata.dashboard.block.container')
                 ->cannotBeEmpty()
             ->end()
+            ->arrayNode('templates')
+                ->addDefaultsIfNotSet()
+                ->children()
+                    ->scalarNode('compose')->defaultValue('SonataDashboardBundle:DashboardAdmin:compose.html.twig')->end()
+                    ->scalarNode('compose_container_show')->defaultValue('SonataDashboardBundle:DashboardAdmin:compose_container_show.html.twig')->end()
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php]: Please linebreak this, and use the `info()` method to provide some doc

File: DependencyInjection/SonataDashboardExtension.php [hunk 5]
```diff
@@ -57,6 +58,9 @@ public function registerParameters(ContainerBuilder $container, array $config)
         $container->setParameter('sonata.dashboard.admin.block.entity', $config['class']['block']);
         $container->setParameter('sonata.dashboard.admin.dashboard.entity', $config['class']['dashboard']);
 
+        $container->setParameter('sonata.dashboard.admin.dashboard.templates.compose', $config['templates']['compose']);
+        $container->setParameter('sonata.dashboard.admin.dashboard.templates.compose_container_show', $config['templates']['compose_container_show']);
```
Comment (greg0ire) [inline DependencyInjection/SonataDashboardExtension.php]: Please linebreak this

File: Resources/config/admin.xml [hunk 6]
```diff
@@ -22,6 +22,12 @@
             <call method="setTranslationDomain">
                 <argument>%sonata.dashboard.admin.dashboard.translation_domain%</argument>
             </call>
+            <call method="setTemplates">
+            	<argument type="collection">
+            		<argument key="compose">%sonata.dashboard.admin.dashboard.templates.compose%</argument>
```
Comment (greg0ire) [inline Resources/config/admin.xml]: Don't use tabs for indention please.

File: DependencyInjection/SonataDashboardExtension.php (hunk 5 above)
Comment (stphnpt) [inline DependencyInjection/SonataDashboardExtension.php]: There's no linebreak for these statements in the other SonataBundles (I try to be consistent with your usual styling)

Comment (greg0ire) [inline DependencyInjection/SonataDashboardExtension.php]: I don't mean linebreak all of them. Linebreak any piece of text as soon as it is longer than 120 chars. This goes for every piece of text, except class declarations and method signatures, to be consistent with Symfony.

**Comment by stphnpt** [comment]
@greg0ire I don't understand the "make: *** [lint] Error 1" for build job 134.9 in the failed checks

**Comment by Soullivaneuh** [comment]
@stphnpt You indentation is not correct by 2 spaces: https://travis-ci.org/sonata-project/SonataDashboardBundle/jobs/136932833#L241

File: DependencyInjection/Configuration.php [hunk 7]
```diff
@@ -46,6 +46,19 @@ public function getConfigTreeBuilder()
                 ->defaultValue('sonata.dashboard.block.container')
                 ->cannotBeEmpty()
             ->end()
+            ->arrayNode('templates')
+                ->addDefaultsIfNotSet()
+                ->children()
+                    ->scalarNode('compose')
+                        ->defaultValue('SonataDashboardBundle:DashboardAdmin:compose.html.twig')
+                        ->info('This value sets the composer template.')
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php]: It is "composer", or should it be "compose"?

Comment (stphnpt) [inline DependencyInjection/Configuration.php]: it corresponds to the "composer" having two templates ("compose" and "compose_container_show") same as for SonataPageBundle

Comment (greg0ire) [inline DependencyInjection/Configuration.php]: Ok then!

**Comment by greg0ire** [comment]
LGTM

File: DependencyInjection/SonataDashboardExtension.php [hunk 8]
```diff
@@ -57,6 +58,11 @@ public function registerParameters(ContainerBuilder $container, array $config)
         $container->setParameter('sonata.dashboard.admin.block.entity', $config['class']['block']);
         $container->setParameter('sonata.dashboard.admin.dashboard.entity', $config['class']['dashboard']);
 
+        $container->setParameter(
+            'sonata.dashboard.admin.dashboard.templates.compose', $config['templates']['compose']);
+        $container->setParameter(
+            'sonata.dashboard.admin.dashboard.templates.compose_container_show', $config['templates']['compose_container_show']);
```
Comment (greg0ire) [inline DependencyInjection/SonataDashboardExtension.php]: Dunno how this passed validation @Soullivaneuh .

@stphnpt this is how it should be done : 

```
$container->setParameter(
    'sonata.dashboard.admin.dashboard.templates.compose',
    $config['templates']['compose']
);
```

Comment (Soullivaneuh) [inline DependencyInjection/SonataDashboardExtension.php]: @stphnpt is PSR compliant, but your sample is better IMHO.

**Comment by Soullivaneuh** [comment]
Sorry for the auto-close. This is because we decide to work on `master` only for now.

Don't worry, to continue the work, simply reopen a new PR with the same branch but against `master` this time. :+1:

**Pull request closed without merging**